### PR TITLE
remove CVEs

### PR DESCRIPTION
**Pull request by hwchase17**
This PR aims to move all code with CVEs into `langchain.experimental`. Note that we are NOT yet removing from the core `langchain` package - we will give people a week to migrate here.

See MIGRATE.md for how to migrate

Zero changes to functionality

Vulnerabilities this addresses:

PALChain:
- https://security.snyk.io/vuln/SNYK-PYTHON-LANGCHAIN-5752409
- https://security.snyk.io/vuln/SNYK-PYTHON-LANGCHAIN-5759265

SQLDatabaseChain
- https://security.snyk.io/vuln/SNYK-PYTHON-LANGCHAIN-5759268

`load_prompt` (Python files only)
- https://security.snyk.io/vuln/SNYK-PYTHON-LANGCHAIN-5725807

### CODE REVIEW

**Comment by hwchase17** [comment]
@boazwasserman @Jflick58 would love thoughts and input here

**Comment by hwchase17** [comment]
merging, but will continue to update

**Comment by boazwasserman** [comment]
@hwchase17 

Great stuff. Definitely a step in the right direction.
The next thing should be adding security layers in the more useful chains (like SQL) to be able to confidently bring them back to the core lib.
I would be happy to contribute to this.

**Comment by orraz-labs** [comment]
@hwchase17 Consider doing the same with the API chain, or perhaps merge some of the changes i've introduced here: https://github.com/hwchase17/langchain/pull/6228/files
Honestly i'm not sure why the pipeline failed there.

**Comment by hwchase17** [comment]
> @hwchase17
> 
> Great stuff. Definitely a step in the right direction. The next thing should be adding security layers in the more useful chains (like SQL) to be able to confidently bring them back to the core lib. I would be happy to contribute to this.

@boazwasserman - would love to see what this would involve. im a bit naive here. another option would be to develop a separate package focused exlcusive on that. we probably wouldnt be able to maintain that, but we would definitely help/contribute if you wanted to!

> @hwchase17 Consider doing the same with the API chain, or perhaps merge some of the changes i've introduced here: https://github.com/hwchase17/langchain/pull/6228/files
> Honestly i'm not sure why the pipeline failed there.

will comment in that pr!

**Comment by Jflick58** [comment]
Wanted to include this conversation here because I think it's more broadly relevant here: 

> sorry for the delay here. high level thoughts:
> 
> 1. We almost certainly don't want to introduce this as a dependency for the main project
> 2. We want to support as flexible of python running for prototyping purposes as possible
> 3. We do not want people to be running this in production
> 
> I would suggest the following changes:
> 
> 1. We will move the current python tools and anything that depends on them into `langchain.experimental` and make that a separate package. This should remove CVEs from core langchain, while also hopefullying making it clear its experimental, and also allowing for prototyping
> 2. I would encourage the creation of a separate `langchain_python` or the like respository where that uses wasm by default. we'd be happy to support this in a variety of ways (help setting up, help publicizing, help maintaining), but i think we'd prefer it to be a separate repository
> 
> thoughts?

Do you want `langchain.experimental` to be an acknowledge of risk "hey these things aren't ready for production, do not rely on it, but useful for prototyping" or to be the "playground" for riskier stuff (main contain CVEs) with a path to merging back into the core repo if the CVEs are resolved?  I think this ultimately comes down to what the path to production looks like with Langchain - is the answer that some things will just need more complicated orchestration to do safely in production, and thus is just something that can't be captured in a standalone library?

**Comment by boazwasserman** [comment]
@Jflick58 @hwchase17 
I think it is possible to add sufficient validation / sandboxing to make dangerous chains significantly safer.
The thing is that we probably won't be able to solve 100% of problematic flows, because many of them actually depend on the context of the application. So the question is with enough mitigation capabilities and controls within langchain, when does it pass the bar to be production ready?

The safest approach for langchain as a library is to never be the one executing the LLM output, and by executing I mean things like running a DB query, sending an HTTP request, executing python code or running a shell command.
That way the responsibility is passed on to the app developer that has to make sure to make the necessary validations needed for his app specific context. If I understood correctly, that is part of the idea of the new modularity approach (https://github.com/langchain-ai/langchain/discussions/7662), where each component can be accessed separately.
My opinion is that adding guardrails and controls on the langchain side and passing the app specific configuration and additional validation to the app developer is the way to go in terms of the main package.

If you agree with the above, we can start by breaking apart some of the dangerous chains that have basic validations (PAL and SQL for example), and introduce them back to the main package using only the generating capability and leaving out the execution capability. Then we can add the validations to the generated output as a generic guardrail for the developer, that can further add validations on their specific app context.

File: MIGRATE.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+Now:
+
+`from langchain.experimental.pal_chain import PALChain`
+
+## SQLDatabaseChain
+
+Previously:
+
+`from langchain.chains import SQLDatabaseChain`
```
Comment (SurenderSingh04) [inline MIGRATE.md:30]: from langchain_experimental.sql import SQLDatabaseChain

File: MIGRATE.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+Now:
+
+`from langchain.experimental.sql import SQLDatabaseChain`
```
Comment (SurenderSingh04) [inline MIGRATE.md:34]: from langchain_experimental.sql import SQLDatabaseChain

**Pull request merged**